trame/layouts/core.py

```diff
@@ -166,6 +167,27 @@ def start_client(**_):
 
         _app.run_server(port=0)
 
+    def add_route(self, name, path, template):
+        _app = tri.get_app_instance()
+        # TODO: Check if route already exists
+        _app.routes.append(
+            {
+                "name": name,
+                "path": path,
+                "component": {
+                    "template": template.html,
```

Comment:
You may want to defer the `template.html` evaluation to later.

Reply:
Maybe you just pass the `template` to template and you do a pass somewhere in layout.start to eval to html any `isinstance(template, AbstractElement)`.